### PR TITLE
[CFP-506] Rework actions from job to workflow

### PR DESCRIPTION
**Pull request by w1stler**
It was causing this error: `[GitHub Action error: "Top level 'runs:' section is required"]`. Not sure if I am able to test github actions locally. ;-)

### CODE REVIEW

**Comment by arturtamborski** [comment]
ill take a look in the evening :)

**Comment by arturtamborski** [comment]
Okay, it took me way longer to look at this, sorry. So there are a few things I've done to this to fix it a bit.

First, I've converted it to composite action. That's the correct way of having this part defined. I've talked with you about reusable workflows, but they should be tackled a bit differently. I'll create a PR to show that.

Second, I've removed actions which don't need to be there (we are better off assuming that python and project requirements are installed). That's because some of these assumptions shouldn't repeat across composite actions.
For example, we can't have a setup-python in every composite action that needs python, because it's needlessly slow and potentially breaking. Instead, if the action needs a specific python version, it should probably just use Docker as the runtime (instead of composite).

What I'm trying to do here with this repo is to have a composable set of actions that relies on shared layout across the projects. One such rule requires that all projects wanting to use this action should store it's backend code in `backend/` directory. That's what this action expects, and that's what I think we should be aiming for.  Notice that other actions are also using hardcoded assumptions (`pack-backend` has a lot of `working-directory: backend/`).

I'm sorry for not noting down all of these requirements / assumptions - it's way more frustrating to work with these unknowns from your point of perspective and I'm sorry for not sharing them with you before.

So to sum it up - we'll have another repo for workflows which will rely on actions in this repo. The workflows will be reusable and will rely on a set of internal project rules, such as layout (`backend/frontend split, use of pipenv, etc`) and thanks to these rules we'll be able to set up new projects more quickly, because they will feel familiar across the organisation.
